### PR TITLE
[Merged by Bors] - Multiarch docker GitHub actions

### LINKED ISSUE #1512: Docker Images for arm64 releases (Raspberry Pi)
## Description
Hello,
Please also consider providing Docker Image releases for arm64/aarch64 platforms along with amd64 Docker Images on Docker Hub.

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

Resolves #1512

## Proposed Changes

- Adds a new docker Github Actions workflow  
- Removes the Dockerhub hook
- Adds a new Dockerfile for use with pre-existing cross-compiled binaries 
- on pushes to `unstable` 
  - builds an ARM64 image and tags it `latest-arm64-unstable`
  - builds an AMD64 image and tags it `latest-amd64-unstable`
  - builds an multiarch image by creating a manifest list referencing the prior two images and tags it `latest-unstable`
- on pushes to `stable` 
  - builds an ARM64 image and tags it `latest-arm64`
  - builds an AMD64 image and tags it `latest-amd64`
  - builds an multiarch image by creating a manifest list referencing the prior two images and tags it `latest`

## Additional Info
- for ARM64, first `cross` is used to cross compile the `lighthouse` and  `lcli` binaries, then `docker buildx` is installed to actually build the docker image for the correct target platform. The image build pretty much just copies the binaries from local into the docker image (thanks @michaelsproul :) )
- The AMD64 and ARM64 builds run in parallel, in total it's been taking around 45mins on a local runner
- This PR does **not** cover version tags on docker images at the moment

### CODE REVIEW

**Review by michaelsproul — COMMENTED** [review]
This is awesome!

I'm inclined to merge soon, although I'd like to see how long it takes to run on Github's runners. Maybe we could temporarily tweak the build rules to run for this PR?

To help performance along I think removing `lcli` is a good idea, and we could maybe even try to merge my PR to reduce compilation time (#1989). I'll work on updating it today so it's ready to go.

File: .github/workflows/docker.yml
Comment (michaelsproul) [inline .github/workflows/docker.yml:59]: (oops, forgot to submit this comment)

I'm confused about what this does. It seems to install some binaries or set up some configuration so that docker can run ARM images, but we don't make use of that functionality do we? Or does `docker buildx build --platform=linux/arm64` require it?

File: .github/workflows/docker.yml
Comment (realbigsean) [reply .github/workflows/docker.yml:40]: I added it for symmetry with the normal `Dockerfile`... do you think we should drop it there as well?

Comment (michaelsproul) [reply .github/workflows/docker.yml:40]: Yeah, I've deleted it in https://github.com/sigp/lighthouse/pull/1989

File: .github/workflows/docker.yml
Comment (realbigsean) [reply .github/workflows/docker.yml:59]: I think we could get away with not using it if we didn't have a `RUN` command in `Dockerfile.cross`. But because of that, we have to run the ARM64 version of the debian image during the build

Reference comment from a buildx issue: https://github.com/docker/buildx/issues/132#issuecomment-522177152

**Comment by realbigsean** [comment]
Removed the `lcli` cross-compilation. I'd also like to test on a Github server but I think the workflow has to be in the default branch before it can be triggered (unless there's some manual workaround?), so don't think I can from my PR here.

File: .github/workflows/docker.yml
Comment (michaelsproul) [inline .github/workflows/docker.yml:52]: This comment is doomed. I was seeing 2 versions of it just now, so I deleted one, and it deleted both :facepalm: 

> I'm confused about what this does. It seems to install some binaries or set up some configuration so that docker can run ARM images, but we don't make use of that functionality do we? Or does docker buildx build --platform=linux/arm64 require it?

- Me

Comment (realbigsean) [reply .github/workflows/docker.yml:52]: 😂  I left a reply -> 

I think we could get away with not using it if we didn't have a  `RUN` command in `Dockerfile.cross`. 
There's a relevant comment from a buildx issue here: https://github.com/docker/buildx/issues/132#issuecomment-522177152

But is that something we could/should get rid of?

Comment (realbigsean) [reply .github/workflows/docker.yml:52]: I was just looking at #2021 and realized if we want to run as non-root we'll probably need to add another `RUN` command too

**Comment by realbigsean** [comment]
I disabled my local runner to test on a Github hosted runner (didn't realize you could use these for free) and kicked off a build on `stable` and `unstable` in my forked repo. These are the two runs:
https://github.com/realbigsean/lighthouse/actions/runs/409396423
https://github.com/realbigsean/lighthouse/actions/runs/409395631

File: .github/workflows/docker.yml
Comment (michaelsproul) [reply .github/workflows/docker.yml:52]: Awesome, yeah, I hadn't made the connection between running `apt-get` and QEMU. All good

**Review by michaelsproul — APPROVED** [review]
Happy to merge once the builds on your repo succeed. Hopefully #1989 will have merged by then too.

**Comment by realbigsean** [comment]
The builds on my repo have succeeded, and I tested out the `stable` and `unstable` images on my pi and my laptop. All's well 👍

**Comment by michaelsproul** [comment]
Awesome!

I think we can merge, and then disable the old docker build once it lands. We'll only update the `latest-unstable` tag, which is unused at the moment anyway.

bors r+

**Review by paulhauner — APPROVED** [review]
Nice! This looked like it would have been super fiddly.

Big benefit to users and devs in this one :rocket:

**Comment by michaelsproul** [comment]
🎉 🎉 🎉

**Pull request closed without merging**
